Build failed for ALT Linux due to -Werror=return-type

<!-- This is a suggested template for your issue. (remove as you check)     -->
<!--   * After crash, always backup and include output.txt in the bug report-->
<!--     from your pioneer config folder, before starting a new game        -->
<!--     (consult FAQ on wiki for info on how to find it).                  -->
<!--   * if graphical: also include opengl.txt from pioneer config dir      -->
<!--   * Please Include version of your last bug-free Pioneer, if possible. -->
<!--   * Has the issue been reported before? Search the issue tracker.      -->
<!--   * Perhaps a screenshot illustrates the bug well?                     -->


### Observed behaviour
Compiler version:
x86_64-alt-linux-g++ (GCC) 12.1.1 20220518 (ALT Sisyphus 12.1.1-alt2)

Because in ALT Linux gcc has the `-Werror=return-type` flag set by default, build error occurs here:
```
.../src/pioneer/src/pigui/PerfInfo.cpp: In member function ‘PiGui::PerfInfo::CounterInfo& PiGui::PerfInfo::GetCounter(CounterType)’:
.../src/pioneer/src/pigui/PerfInfo.cpp:115:1: error: control reaches end of non-void function [-Werror=return-type]
  115 | }
      | ^
cc1plus: some warnings being treated as errors
```

the default is a warning, but the situation borders on UB.
In the distribution, this flag was set due to several cases, for example [this](https://git.altlinux.org/gears/s/synaptic.git?p=synaptic.git;a=blob;f=synaptic/gtk/rguserdialog.cc;h=83349c1e64c61013dac72fe293a8055d84558b36;hb=cfd2d6d12072ecb8bd1c38cdd1cdf39c71e1ba28#l211).  There were troubles due to such non-void functions that did not return a value, when changing the compiler version, new synaptic began to crash due to double-free.

EDIT:
Here is some more information about it:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=87515

__My pioneer version (and OS):__
20230203

--- a/src/graphics/opengl/RenderTargetGL.cpp
+++ b/src/graphics/opengl/RenderTargetGL.cpp
@@ -15,6 +15,8 @@ namespace Graphics {
 				case RenderTarget::READ: return GL_READ_FRAMEBUFFER;
 				case RenderTarget::DRAW: return GL_DRAW_FRAMEBUFFER;
 				case RenderTarget::BOTH: return GL_FRAMEBUFFER;
+				// default value is never reached, calm down -Werror=return-type
+				default: return 42;
 			}
 		}
 

--- a/src/pigui/PerfInfo.cpp
+++ b/src/pigui/PerfInfo.cpp
@@ -116,7 +116,8 @@ PerfInfo::CounterInfo &PerfInfo::GetCounter(CounterType ct)
 	case COUNTER_FPS: return m_fpsCounter;
 	case COUNTER_PHYS: return m_physCounter;
 	case COUNTER_PIGUI: return m_piguiCounter;
-	default: assert(0);
+	// default value is never reached, calm down -Werror=return-type
+	default: return m_fpsCounter;
 	}
 }
 

